src/core/Global.ttslua

```diff
@@ -600,41 +618,85 @@ local source_repo = 'https://raw.githubusercontent.com/seth-sced/loadable-object
 local library = nil
 local request_obj
 
-function onClick_toggleUi(player, window)
-  toggle_ui(window)
-end
-
 function onClick_refreshList()
   local request = WebRequest.get(source_repo .. '/library.json', completed_list_update)
   request_obj = request
-  startLuaCoroutine(Global, 'my_coroutine')
+  startLuaCoroutine(Global, 'downloadCoroutine')
 end
 
 function onClick_select(player, params)
   params = JSON.decode(urldecode(params))
   local url = source_repo .. '/' .. params.url
   local request = WebRequest.get(url, function (request) complete_obj_download(request, params) end )
   request_obj = request
-  startLuaCoroutine(Global, 'my_coroutine')
+  startLuaCoroutine(Global, 'downloadCoroutine')
 end
 
 function onClick_load()
   UI.show('progress_display')
   UI.hide('load_button')
 end
 
-function toggle_ui(title)
-  UI.hide('load_ui')
-  if UI.getValue('title') == title or title == 'Hidden' then
-    UI.setValue('title', 'Hidden')
+function onClick_defaultSettings()
+  print("Dummy: Load default settings")
+end
+
+function onClick_toggleOption(_, id)
+  local state = self.UI.getAttribute("toggle" .. id, "isOn")
+
+  -- flip state (and handle stupid "False" value)
+  if state == "False" then
+    state = true
   else
+    state = false
+  end
+
+  self.UI.setAttribute("toggle" .. id, "isOn", state)
+  optionPanel[id] = state
+
+  applyChange(id, state)
+end
+
+local PlayerMatAPI = require("playermat/PlaymatApi")
+function applyChange(id, state)
+  -- option 1: Snap tags
+  if id == "1" then
+    printToAll("Playermat snap tags " .. (state and "en" or "dis") .."abled.", "White")
```

Comment:
I don't think this is necessary.  The user clicked the value.  The value updated.  They can see it :)